overlord/state/taskrunner.go

```diff
@@ -84,6 +84,15 @@ func (r *TaskRunner) AddHandler(kind string, do, undo HandlerFunc) {
 	r.handlers[kind] = handlerPair{do, undo}
 }
 
+// KnownTaskKinds returns all tasks kinds handled by this runner.
+func (r *TaskRunner) KnownTaskKinds() []string {
+	kinds := make([]string, 0, len(r.handlers))
+	for h := range r.handlers {
```

Comment:
Is it possible to have more than one task of given kind?

Reply:
No, this is not about tasks, it's about their kinds (= names of handlers). They are unique.